lift/Readme.md

```diff
@@ -2,275 +2,424 @@
+#### `unapplyMigration`
 
-### Up & Down in migration scripts
+Unapplies the previous Step in the migrations folder. This is called by the Down method in the Lift Client.
 
-In order to revert a migration in the case of a rollback, we need to distinguish between `up` (applying the migration) and `down` (reverting the migration). We
-call this the **Migration Direction**. The direction of a migration is being passed in to a migration script using the env var `DIRECTION`. This can then e.g.
-be accessed from a bash script with `echo $DIRECTION`.
+#### `calculateDatamodel`
 
-### Transactional/rollback behavior of migration scripts
+Used to render the resulting datamodel into the Readme of the migration folder
 
-The main question is this: If a migration script fails (exists with a non-zero exit code) after the datamodel has been migrated, should the datamodel be rolled
-back to the state before? The answer is yes. If you don't want the datamodel migration to be rolled back, make sure, that your migration script will not return
-a non-zero exit code, by e.g. using `try` `catch` in Node.js or adding an `|| echo ""` behind the potentially failing command.
+**TODO:** Double-check with Tim to see if this is still necessary
+
+#### `calculateDatabaseSteps`
+
+Calculate the database steps when a certain migration has not been executed yet. It answers the question:
 
-### Attaching migration information (`rename` & `migrationValue`)
+- What would the database steps be if the we assume that the migration steps have been applied already? This can happen when you have multiple unapplied
+  migrations after calling the Save method multiple times before calling Up.
+
+**TODO:** Double-check with Tim to see if this is still necessary. This may not be needed anymore as lots of assumed steps is also available in
+`inferMigrationSteps` now.
+
+#### `listMigrations`
+
+Lists the migrations we've currently applied to the datasources.
+
+#### `migrationProgress`
+
+Migrations can take a long time to complete. `migrationProgress` returns the progress of the currently running migration.
+
+## Prisma CLI
+
+### Lift subcommands
+
+Prisma users interact with Lift mostly thought Prisma CLI, where there is a collection of commands available beneath `prisma2 lift`:
+
+#### `prisma2 lift save`
+
+Create a new Migration 
+
+- Optionally supply a `--name` to give the migration a name.
+
+#### `prisma2 lift up`
+
+Migrate your database up
+
+Options:  
+- Use `-p`/`--preview` to preview the migration without migrating
+- Use `--auto-approve` to skip interactive approval before migrating
+
+Arguments:  
+- Append a number to go up by x migrations
+- Append a timestamp to go up by timestamp of the migration
+- Append a string to go up by name of the migration
+
+#### `prisma2 lift down`
+
+Migrate your database down
+
+Same options and arguments as `prisma2 lift up`
+
+### Other Commands
+
+Lift functionality is also part of other commands:
+
+#### `prisma2 dev`
+
+TODO
```

Comment:
Can someone suggest a wording of what the Lift parts of `prisma2 dev` actually are and how they work?

Reply:
Ping @timsuchanek @tomhoule